### PR TITLE
Tests for joining with custom content

### PR DESCRIPTION
**Pull request by Kegsay**
My first sytest :) (Thanks @erikjohnston for the help!)

### CODE REVIEW

File: tests/10apidoc/33room-members.pl [hunk 1]
```diff
@@ -148,6 +148,78 @@ sub matrix_join_room
...
+         method => "POST",
+         uri    => "/r0/join/$room_id",
+
+         content => { "foo" => "bar" },
+      )->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( room_id ));
+         $body->{room_id} eq $room_id or
+            die "Expected 'room_id' to be $room_id";
```
Comment (NegativeMjark) [inline tests/10apidoc/33room-members.pl]: Maybe use ``assert_eq`` here.

Comment (Kegsay) [inline tests/10apidoc/33room-members.pl]: done

**Comment by NegativeMjark** [comment]
LGTM

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Pull request merged**